packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java

```diff
@@ -891,6 +893,14 @@ private Application getApplication() {
     }
   }
 
+  private void destroyMapViewIfNecessary() {
+    if (mapViewDestroyed || mapView == null) {
+      return;
+    }
+    mapView.onDestroy();
+    mapViewDestroyed = true;
```

Comment:
what about just making `mapView = null;` ?

Reply:
ditto, can't the state tracking be done implicitly with the mapView instance variable?

Reply:
The NPE didn't happen on the mapView object, it is rather something inside map SDK had a NPE. 
By looking at the log: 
```
Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'void com.google.maps.api.android.lib6.impl.bp.q()' on a null object reference
```
Probably the first mapView.onDestroy cleaned up some google map sdk internal resources and the second call tries the clean up the same resource again and got a NPE.
I also looked up the documentation and I don't find anything that indicates if MapView was destroyed.

Reply:
ya, I got that part. What we're saying is now that we made mapView unusable by calling onDestroy on it, let's just remove its reference. I don't see that done anywhere in the GoogleMapController. If you did, you wouldn't have to use a separate bool to track whether mapView is destroyed.

Reply:
Ah I see. I was considering it but I wasn't sure because it'll remove the `final` on mapView.

Reply:
sure, you'd have to edit the doc for getView() too. But maybe that's a feature rather than a 'bug' since it's weird to start with that you could have a public API to 'get' a destroyed view.

Reply:
Sounds good. Will update that! thanks.